proxy/src/auth/backend/mod.rs

```diff
@@ -282,8 +282,8 @@ async fn auth_quirks(
         Ok(info) => (info, None),
     };
 
-    debug!("fetching user's authentication info");
-    let (allowed_ips, maybe_secret) = api.get_allowed_ips_and_secret(ctx, &info).await?;
+    debug!("fetching authentication info and allowlists");
+    let allowed_ips = api.get_allowed_ips(ctx, &info).await?;
```

Comment:
nit: should we only do this if `config.ip_allowlist_check_enabled` is true?

Reply:
yes, agree